Framework/NexusGeometry/test/NexusGeometrySaveTest.h

```diff
@@ -0,0 +1,1539 @@
+// Mantid Repository : https://github.com/mantidproject/mantid
+//
+// Copyright &copy; 2019 ISIS Rutherford Appleton Laboratory UKRI,
+//     NScD Oak Ridge National Laboratory, European Spallation Source
+//     & Institut Laue - Langevin
+// SPDX - License - Identifier: GPL - 3.0 +
+#ifndef MANTID_NEXUSGEOMETRY_NEXUSGEOMETRYSAVETEST_H_
+#define MANTID_NEXUSGEOMETRY_NEXUSGEOMETRYSAVETEST_H_
+
+#include "MantidGeometry/Instrument/ComponentInfo.h"
+#include "MantidGeometry/Instrument/DetectorInfo.h"
+#include "MantidGeometry/Instrument/InstrumentVisitor.h"
+#include "MantidKernel/EigenConversionHelpers.h"
+#include "MantidKernel/ProgressBase.h"
+#include "MantidKernel/WarningSuppressions.h"
+#include "MantidNexusGeometry/NexusGeometrySave.h"
+#include "MantidTestHelpers/ComponentCreationHelper.h"
+
+#include <H5Cpp.h>
+#include <algorithm>
+#include <boost/filesystem.hpp>
+#include <boost/make_shared.hpp>
+#include <cxxtest/TestSuite.h>
+#include <fstream>
+#include <gmock/gmock.h>
+#include <iostream>
+#include <memory>
+#include <set>
+
+#define DEBUG_PRINT(x) std::cout << "\nDEBUG: " << x << std::endl
+
+using namespace Mantid::NexusGeometry;
+
+//---------------------------------------------------------------
+namespace {
+
+typedef std::vector<std::string> fullH5Path;
+
+// get path as string. Used for the dependency tests.
+std::string toH5PathString(fullH5Path &path) {
+  std::string pathString = "";
+  for (const std::string &grp : path) {
+    pathString += "/" + grp;
+  }
+  return pathString;
+}
+
+const H5G_obj_t GROUP_TYPE = static_cast<H5G_obj_t>(0);
```

Comment:
Is this the third duplication? See comment above about creating `NexusDefinitions` and using here directly.

Reply:
done